### PR TITLE
Minor refactorings

### PR DESCRIPTION
**Pull request by jechol**
Minor refactorings.

Please see logs for each commit.

### CODE REVIEW

**Comment by srdanrasic** [comment]
Thank you @jechol, I appreciate this.

File: Sources/RawStream.swift [hunk 1]
```diff
@@ -106,24 +99,21 @@ public extension _StreamType {
...
-          if !disposable.isDisposed {
-            observer.next(number)
-            number += 1
-            dispatch()
-          } else {
-            dispatch = nil
-          }
+          guard !disposable.isDisposed else { dispatch = nil; return }
+          observer.next(number)
+          number = number + 1
```
Comment (srdanrasic) [inline Sources/RawStream.swift]: I'm used to `number += 1` but it's not that important.

File: Sources/RawStream.swift [hunk 2]
```diff
@@ -56,13 +56,6 @@ extension _StreamType {
 /// Represents an underlying stream generalized over EventType and used by
 /// higher-level implementations like `Stream<T>` or `Operation<T, E>`.
 public protocol RawStreamType: _StreamType {
-
-  /// The type of events generated by the stream.
-  associatedtype Event: EventType
-
-  /// Register an observer that will receive events from a stream.
-  @warn_unused_result
-  func observe(observer: Event -> Void) -> Disposable
```
Comment (srdanrasic) [inline Sources/RawStream.swift]: I had a problem with this not being redeclared here, but it seems to be working now.

File: Sources/RawStream.swift (hunk 1 above)
Comment (jechol) [inline Sources/RawStream.swift]: FYI, `number += 1` doesn't work for IntegerType.

File: Sources/Threading.swift [hunk 3]
```diff
@@ -101,9 +101,21 @@ public extension Queue {
...
 }
 
+internal protocol Lock {
+  func lock()
+  func unlock()
+  func atomic<T>(@noescape body: () -> T) -> T
+}
+
+internal extension Lock {
+  func atomic<T>(@noescape body: () -> T) -> T {
```
Comment (srdanrasic) [inline Sources/Threading.swift]: 👍

File: Sources/RawStream.swift (hunk 1 above)
Comment (srdanrasic) [inline Sources/RawStream.swift]: Ah, ok than.

**Pull request merged**